app/src/parser_impl.c

```diff
@@ -864,6 +864,103 @@ parser_error_t _readBoxes(parser_context_t *c, box boxes[], uint8_t *num_element
     return parser_ok;
 }
 
+__Z_INLINE parser_error_t _readHoldingElement(parser_context_t *c, holding *holding)
+{
+    uint8_t key[2] = {0};
+    uint16_t mapSize = 0;
+    CHECK_ERROR(_readMapSize(c, &mapSize))
+    holding->d = 0;
+    holding->s = 0;
+
+    for (uint16_t index = 0; index < mapSize; index++) {
+        CHECK_ERROR(_readString(c, key, sizeof(key)))
+        if (strncmp((char *)key, KEY_APP_ADDRESS, sizeof(KEY_APP_ADDRESS)) == 0) {
+            CHECK_ERROR(_readUInt8(c, &holding->d));
+
+        } else if (strncmp((char *)key, KEY_APP_ASSET, sizeof(KEY_APP_ASSET)) == 0) {
+            CHECK_ERROR(_readUInt8(c, &holding->s));
+        } else {
+            return parser_unexpected_error;
+        }
```

Comment:
I'd advocate for using `switch` rather than `if-else` logic here. But moreover I'm wondering if we could avoid using `strncmp` for single-character comparisons. Maybe something like this would address both:
```suggestion
    switch (key[0]) {
      case KEY_APP_ADDRESS[0]:
        CHECK_ERROR(_readUInt8(c, &holding->d));
        break;
      case KEY_APP_ASSET[0]:
        CHECK_ERROR(_readUInt8(c, &holding->s));
        break;
      default:
        return parser_unexpected_error;
    }
```
Same applies to `_readLocalElement` too.